ballerina/SQLClient.bal

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/sql;
+
+public client class SQLClient {
+
+    private final sql:Client dbClient;
+
+    private string entityName;
+    private sql:ParameterizedQuery tableName;
+    private map<FieldMetadata> fieldMetadata;
+    private string[] keyFields;
+
+    public function init(string entityName, sql:ParameterizedQuery tableName, map<FieldMetadata> fieldMetadata, string[] keyFields, sql:Client dbClient) returns error? {
+        self.entityName = entityName;
+        self.tableName = tableName;
+        self.fieldMetadata = fieldMetadata;
+        self.keyFields = keyFields;
+        self.dbClient = dbClient;
+    }
+
+    function runInsertQuery(record {} 'object) returns sql:ExecutionResult|error {
```

Comment:
we need to make these functions public to access in generated clients

Reply:
Fixed in [7600af1](https://github.com/ballerina-platform/module-ballerina-persist/pull/1/commits/7600af1975f3074d3823f5aa3989289aba4dfe3e)